packages/react-components/source/react/internal/option-menu-list/OptionMenuList.js

```diff
@@ -17,21 +17,25 @@ import {
 import OptionMenuListItem from './OptionMenuListItem';
 import Icon from '../../library/icon';
 
+export const valueType = PropTypes.oneOfType([
+  PropTypes.string,
+  PropTypes.shape({
+    key: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+  }),
```

Comment:
is valueType allowed to be null? If not, we should add `.isRequired` to the end of for this declaration.

Also, `.isRequired` ought to be added to the end of the `key` I think.